studymanager/src/test/java/io/redlink/more/studymanager/controller/studymanager/ParticipantControllerTest.java

```diff
@@ -61,6 +65,13 @@ void setUp() {
                         EnumSet.allOf(PlatformRole.class)
                 )
         );
+
+        when(gatewayProperties.baseUrl()).thenReturn(signupUrl);
+        when(gatewayProperties.generateSignupUrl(any(Participant.class))).thenAnswer(invocation -> {
+            Participant participant = invocation.getArgument(0);
+            if (participant.getRegistrationToken() == null) return null;
+            return URI.create(signupUrl + "/signup?token=" + participant.getRegistrationToken());
+        });
```

Comment:
You're mocking an important part of the implementation to test.

To use the `GatewayProperties` in your tests, just eject them as a Bean in a `TestConfiguration` to have it picked up by the spring-magic:

```java
    @TestConfiguration
    static class TestConfig {
        @Bean
        GatewayProperties gatewayProperties() {
            return new GatewayProperties(SIGNUP_URL);
        }
    }    
```

Reply:
Done in https://github.com/MORE-Platform/more-studymanager-backend/pull/361/commits/68f46b2b29ca2cd9111cad67bed8c2b6428b4e5d